src/hessian.jl

```diff
@@ -1,78 +1,117 @@
-# Tags:
-Cassette.@context HessianSparsityContext
+# TermCombination datastructure
 
-const TaggedOf{T} = Tagged{A, T} where A
+struct TermCombination
+    terms::Set{Dict{Int, Int}} # idx => pow
+end
 
-const HTagType = Union{Input, TermCombination}
-Cassette.metadatatype(::Type{<:HessianSparsityContext}, ::DataType) = HTagType
+@eval Base.zero(::Type{TermCombination}) = $(TermCombination(Set{Dict{Int,Int}}()))
```

Comment:
what's the reason for this being an eval?

Reply:
So that it's not creating a Set every time it's called. This creates a singleton. All operations on TermCombination are pure.